app/core/lifespan.py

```
@@ -99,6 +200,39 @@ async def create_lifespan(app: FastAPI) -> AsyncIterator[None]:
             pass
         logger.info("Chain anchor task stopped")
 
+    tsa_batch_task = getattr(app.state, "tsa_batch_task", None)
+    if tsa_batch_task is not None:
+        tsa_batch_task.cancel()
+        try:
+            await tsa_batch_task
+        except asyncio.CancelledError:
+            pass
```

Ruff (0.15.5)

[warning] Use `contextlib.suppress(asyncio.CancelledError)` instead of `try`-`except`-`pass`

Replace `try`-`except`-`pass` with `with contextlib.suppress(asyncio.CancelledError): ...`

(SIM105)